app/controllers/account/users_controller.rb

```diff
@@ -1,19 +1,50 @@
 # frozen_string_literal: true
 module Account
   class UsersController < ApplicationController
+    before_action :authenticate_user!
+    before_action :set_user, only: [:destroy]
```

Comment:
It is a bad practice to define instance variable in private method.
please remove this before action, rename set_user to resource, and add @user = resource in destroy method and everywhere where you need @user.